doc/nrf/protocols/matter/overview/integration.rst

```diff
@@ -91,3 +91,28 @@ For this design, the Wi-Fi driver on the application core communicates with the
    :alt: Multiprotocol Wi-Fi and Bluetooth LE architecture (nRF53 with the nRF7002 Wi-Fi 6 Companion IC)
 
    Multiprotocol Wi-Fi and Bluetooth LE architecture (nRF53 with the nRF7002 Wi-Fi 6 Companion IC)
+
+Switchable Matter over Wi-Fi and Matter over Thread
+===================================================
+
+In this design, the Matter stack, the OpenThread stack, Wi-Fi stack, and the Bluetooth LE stack run on a single SoC.
```

Comment:
```suggestion
In this design, the Matter stack, the OpenThread stack, Wi-Fi stack, and the Bluetooth LE stack run on a single SoC. Due to internal flash size limitation, only one transport protocol can be used at a time. User may decide to switch between Matter over Thread and Matter over Wi-Fi images during the boot of the device. In case image has to be swapped, the new image is loaded from the external flash to the internal flash and run.
```

Reply:
I've only added the first new sentence. All other information is covered by the include from the lock sample.